libr/anal/sign.c

```diff
@@ -67,6 +67,44 @@ R_API RList *r_sign_fcn_vars(RAnal *a, RAnalFunction *fcn) {
 	return ret;
 }
 
+R_API RList *r_sign_fcn_types(RAnal *a, RAnalFunction *fcn) {
+
+	// From anal/types/*:
+	// Get key-value types from sdb matching "func.%s", fcn->name
+	// Get func.%s.args (number of args)
+	// Get type,name pairs
+	// Put everything in RList following the next format:
+	// types: main.ret=%type%, main.args=%num%, main.arg.0="int,argc", ...
+
+	r_return_val_if_fail (a && fcn, NULL);
+
+	RList *ret = r_list_newf ((RListFree) free);
+	SdbKv *kv = NULL;
+	SdbListIter *it = NULL;
+	char *args_expr = r_str_newf ("func.%s.args", fcn->name);
+	SdbList *fcntypes = sdb_foreach_match (a->sdb_types, args_expr, 0);
```

Comment:
Wouldn't a single `sdb_get_const()` work here instead of the match, iterating through the list and strcmp?

Reply:
Yes, looking at my code now I see you are right. I don't even know why I did it this way to begin with, lol